modules/core/src/main/java/org/apache/ignite/internal/causality/VersionedValue.java

```diff
@@ -594,7 +594,7 @@ private void trimToSize(long causalityToken) {
      * @param causalityToken Causality token.
      */
     private static void checkToken(long actualToken, long causalityToken) {
```

Comment:
Not related to proposed changes but...
Both actualToken and causalityToken are causality tokens, I'd rather use "appliedToken" and "candidateToken" or similar. If you don't mind, lets fix it here. Besides that checkToken is rather assertToken, up to you though.

Reply:
WDYT about (actualToken, candidateToken)?